scripts/ExternalInterface.lua

```diff
@@ -218,20 +218,105 @@ function AutoDrive:getCombineOpenPipePercent(combine)	--for AIVE
 	return pipePercent
 end
 
--- start CP at first wayPoint
-function AutoDrive:StartCP(vehicle)
-    if vehicle == nil then 
-        return 
+-- start CP using CP HUD settings. Allows CP to control job parameters.
+function AutoDrive:StartCP(startCP_Vehicle)
+    local cpDelay = AutoDrive.getSetting("CPDelayWaitTime", startCP_Vehicle)-- delay in seconds, set in vehicle settings.
+    local cpDelayInc = 0.35 -- timer addition to make it work for seconds as cpDelay
+    local isPassingToCP = (startCP_Vehicle.ad.stateModule:getStartCP_AIVE() and startCP_Vehicle.ad.stateModule:getUseCP_AIVE())
+    local cpStartMarker = startCP_Vehicle.ad.stateModule:getFirstMarker().name
+    local mapMarker = ADGraphManager:getMapMarkerByWayPointId(startCP_Vehicle.destinationID)
+    if startCP_Vehicle.cpDelayWait == false then
+        AutoDrive.debugPrint(startCP_Vehicle, AutoDrive.DC_EXTERNALINTERFACEINFO, "AutoDrive:StartCP...")
+
```

Comment:
What about refilling ?
If you let AD drive to the field and let CP start on the 1st Waypoint, what happens if he comes back ?
There is already a correct solution for this.
For lane offset, we did a band aid in 7.0.0.10

Reply:
I did not touch the portion of AD that uses the restartCP variable. I will test vehicle refuel during CP work on my branch tonight if I get a chance.